llmfoundry/models/layers/attention.py

```diff
@@ -177,7 +177,7 @@ def flash_attn_fn(
     try:
         from flash_attn import bert_padding, flash_attn_interface  # type: ignore # yapf: disable # isort: skip
     except:
-        raise RuntimeError('Please install flash-attn==1.0.3.post0')
+        raise RuntimeError('Please install flash-attn via pip install .[gpu]')
```

Comment:
This tells the user to run the install instructions in README.md
But if they are running a file from `llm-foundry/scripts`, then the correct instruction might be `pip install ..[gpu]`?
Can we just direct the user to follow the install instructions in the readme?

Reply:
We use `pip install .[gpu]` elsewhere in the file, e.g. see https://github.com/mosaicml/llm-foundry/blob/main/llmfoundry/models/layers/attention.py#L309

Given that these errors are being raised by the `llmfoundry` package, I think the right message is to do `pip install llmfoundry[gpu]` since users may not even have our repository cloned. 

Lets keep the message for now, and I'll do a follow-on PR to fix this issue globally.

Reply:
I think this is a bit more complicated because pretty much all of our users use llm foundry directly (can look at pypi download stats) as opposed to as a pip package. Add the pip package is missing a couple dependencies due to our custom installs of a few things (https://github.com/mosaicml/llm-foundry/blob/dc1374816a007488ca9885c92e83e89e6fe933ef/.github/workflows/release.yaml#L39-L41). But definitely not needed to resolve on this PR.

Reply:
ah sorry. this statement is actually for anyone using MPT from huggingface, so it was originally written to be independent of usage in llmfoundry. and to resolve it, only flash attn as specified is needed, not all of llmfoundry. so maybe we leave it?

Reply:
@hanlint could you change this one back? Its actually kind of important for anyone using mpt not inside of llm foundry.

Reply:
OK sure will revert